pkg/asset/installconfig/gcp/gcp.go

```diff
@@ -1,10 +1,118 @@
 package gcp
 
 import (
+	"context"
+	"fmt"
+	"time"
+
 	"github.com/openshift/installer/pkg/types/gcp"
+	"github.com/openshift/installer/pkg/types/gcp/validation"
+	"github.com/pkg/errors"
+	compute "google.golang.org/api/compute/v1"
+	"google.golang.org/api/option"
+	"gopkg.in/AlecAivazis/survey.v1"
 )
 
 // Platform collects GCP-specific configuration.
 func Platform() (*gcp.Platform, error) {
-	return &gcp.Platform{}, nil
+	project, err := selectProject()
+	if err != nil {
+		return nil, err
+	}
+
+	region, err := selectRegion(project)
+
+	return &gcp.Platform{
+		ProjectID: project,
+		Region:    region,
+	}, nil
+}
+
+func getComputeService(ctx context.Context) (svc *compute.Service, err error) {
+	ssn, err := GetSession(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to establish GCP session")
+	}
+
+	svc, err = compute.NewService(ctx, option.WithCredentials(ssn.Credentials))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to initialize GCP compute client")
+	}
+	return svc, nil
+}
+
+func selectProject() (selectedProject string, err error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), 1*time.Minute)
```

Comment:
TODO makes sense when a function cannot be changed to add the context in it's fingerprint. New functions shouldn't have TODO.